storage/innobase/handler/ha_innodb.cc

```diff
@@ -14302,61 +14285,50 @@ ha_innobase::records_in_range(
 				    + sizeof(dtuple_t)));
 
 	range_start = dtuple_create(heap, key->ext_key_parts);
-	dict_index_copy_types(range_start, index, key->ext_key_parts);
 
 	range_end = dtuple_create(heap, key->ext_key_parts);
-	dict_index_copy_types(range_end, index, key->ext_key_parts);
-
-	row_sel_convert_mysql_key_to_innobase(
-		range_start,
-		m_prebuilt->srch_key_val1,
-		m_prebuilt->srch_key_val_len,
-		index,
-		(byte*) (min_key ? min_key->key : (const uchar*) 0),
-		(ulint) (min_key ? min_key->length : 0));
-
```

Comment:
For both min_key and max_key we were calling row_sel_convert_mysql_key_to_innobase with key_ptr=nullptr. While the function row_sel_convert_mysql_key_to_innobase() doesn't seem to handle nullptr and is likely was an issue already.

This is great that the refactoring change improves this part. It also indicates that we may not have any mtr test covering the case when either of min or max key is zero. Can we check if such tests are there or can be added ?

Reply:
That function definitely did receive `key_ptr=nullptr`, but GCC does not care about the UB of adding something to a null pointer. Only `clang` does, but unfortunately `clang -fsanitize=undefined`  is still unusable due to bugs in code outside InnoDB. I got it flagged in GCC after I added the `__attribute__((nullptr))`.  I wrote about that in the thread about the first comment of your first review. That is what prompted this entire change.